onnxruntime/core/framework/session_options.cc

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/framework/session_options.h"
+#include "core/common/logging/logging.h"
+
+namespace onnxruntime {
+
+bool HasSessionConfigEntry(const SessionOptions& options, const std::string& config_key) {
+  return options.session_configurations.find(config_key) != options.session_configurations.cend();
+}
+
+bool AddSessionConfigEntryImpl(SessionOptions& options, const char* config_key, const char* config_value) {
+  std::string key(config_key);
+  if (key.empty() || key.length() > 128) {
+    LOGS_DEFAULT(ERROR) << "config_key is empty or longer than maximum length 128";
+    return false;
+  }
```

Comment:
Also return a status with the error message so it's able to be directly returned from the public API. A log message may not be noticed leading to a support request.